code/ui/generalhud/chat/ChatBox.cs

```diff
@@ -126,48 +129,63 @@ private void Submit()
             }
         }
 
+        public void AddServerInfoEntry (string message, string avatar)
+        {
+            ChatEntry serverEntry = new ChatEntry(message);
+            serverEntry.AddClass("serverInfo");
+
+            _canvas.AddChild(serverEntry);
+        }
         public void AddEntry(string name, string message, string avatar, LifeState lifeState, string team = null)
         {
             _lastChatFocus = 0f;
 
-            if (Messages.Count > MAX_MESSAGES_COUNT)
-            {
-                ChatEntry entry = Messages[0];
+            // if(string.IsNullOrEmpty(name)) {
```

Comment:
Commented out code without any explanation is probably not needed.
You can delete it or add a comment explaining why it needs to stay in.